main/src/OnACoffeeLoop.java

```diff
@@ -0,0 +1,214 @@
+        } else if (fill.equals(actionType)) {
+            fill();
+        } else if (take.equals(actionType)) {
+            take();
+        } else if (remaining.equals(actionType)) {
+            remaining();
+        }
+        else if(exit.equals(actionType))
+        {
+            exit();
+        }
+    }
+
+    public static void buy ()   //method buy asks to input which coffee wants to make,or if change the mind type back
+    {
+        System.out.println("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:");
+        Scanner scaner = new Scanner(System.in);
+        String coffeeType = scaner.nextLine();
+        if(espresso.equals(coffeeType))
+        {
+            espresso();
+        }
+        else if (cappuccino.equals(coffeeType))
+        {
+            cappuccino();
+        }
+        else if (latte.equals(coffeeType))
+        {
+            latte();
+        }
+        else if (back.equals(coffeeType))
+        {
+            action();
+        }
+
+    }
+
+    public static void espresso ()          //method espresso which checks if available ingredients, makes a coffee,
+    //if not asks user what part of ingredient is missing:
+    {
+        if(water>=250 && coffeeBeans>=16 && disCups!=0) {
+            System.out.println("I have enough resources, making you a coffee!");
+            water -=250;
+            coffeeBeans -=16;
+            money +=4;
+            disCups-=1;
+            action();
+        }
+        else if(water<250)
+        {
+            System.out.println("Sorry, not enough water!");
+        }
+        else if(coffeeBeans<16)
+        {
+            System.out.println("Sorry, not enough Coffee Beans!");
+        }
+        else if(disCups==0)
+        {
+            System.out.println("Sorry, not enough disposable cups");
+
+        }
+        action();
+
+    }
+
+    public static void cappuccino ()       //method cappuccino which checks if available ingredients, makes a coffee,
+    //if not asks user what part of ingredient is missing:
+    {
+        if(water>=200 && milk>=100 && coffeeBeans>=12 && disCups!=0) {
+            System.out.println("I have enough resources, making you a coffee!");
+            water -= 200;
+            milk -= 100;
+            coffeeBeans -= 12;
+            money += 6;
+            disCups -= 1;
+            action();
+        }
+        else if(water<200)
+        {
+            System.out.println("Sorry, not enough water!");
+        }
+        else if(milk<100)
+        {
+            System.out.println("Sorry, not enough milk!");
+        }
+        else if(coffeeBeans<12)
+        {
+            System.out.println("Sorry, not enough coffee beans!");
+        }
+        else if(disCups==0)
+        {
+            System.out.println("Sorry, not enough disposable cups");
+        }
+        action();
+    }
+    public static void latte ()     //method latte which checks if available ingredients, makes a coffee,
```

Comment:
@Edvard777 will be better to have comments on top of the method with /** comment **/

Reply:
done